examples/button/js/button.js

```diff
@@ -2,84 +2,114 @@
 *   This content is licensed according to the W3C Software License at
 *   https://www.w3.org/Consortium/Legal/2015/copyright-software-and-document
 *
-*   File:   button.js
-*
-*   Desc:   JS code for Button Design Pattersn
+*   JS code for the button design pattern
 */
 
-var ICON_MUTE_URL  = 'images/mute.svg#icon-mute';
+var ICON_MUTE_URL = 'images/mute.svg#icon-mute';
 var ICON_SOUND_URL = 'images/mute.svg#icon-sound';
 
 function init () {
-  // Create variables for the various buttons
   var actionButton = document.getElementById('action');
-  var toggleButton = document.getElementById('toggle');
-
-  // Add event listeners to the various buttons
-  actionButton.addEventListener('click', actionButtonEventHandler);
-  actionButton.addEventListener('keydown', actionButtonEventHandler);
-
-  toggleButton.addEventListener('click', toggleButtonEventHandler);
-  toggleButton.addEventListener('keydown', toggleButtonEventHandler);
+  actionButton.addEventListener('click', activateActionButton);
+  actionButton.addEventListener('keydown', actionButtonKeydownHandler);
+  actionButton.addEventListener('keyup', actionButtonKeyupHandler);
 
+  var toggleButton = document.getElementById('toggle');
+  toggleButton.addEventListener('click', toggleButtonClickHandler);
+  toggleButton.addEventListener('keydown', toggleButtonKeydownHandler);
+  toggleButton.addEventListener('keyup', toggleButtonKeyupHandler);
 }
 
-function actionButtonEventHandler (event) {
-  var type = event.type;
-
-  // Grab the keydown and click events
-  if (type === 'keydown') {
-    // If either enter or space is pressed, execute the funtion
-    if (event.keyCode === 13 || event.keyCode === 32) {
-      window.print();
-
-      event.preventDefault();
-    }
+/**
+ * Activates the action button with the enter key.
+ *
+ * @param {KeyboardEvent} event
+ */
+function actionButtonKeydownHandler (event) {
+  // The action button is activated by space on the keyup event, but the
+  // default action for space is already triggered on keydown. It needs to be
+  // prevented to stop scrolling the page before activating the button.
+  if (event.keyCode === 32) {
+    event.preventDefault();
+  }
+  // If enter is pressed, activate the button
+  else if (event.keyCode === 13) {
+    event.preventDefault();
+    activateActionButton();
   }
-  else if (type === 'click') {
-    window.print();
+}
+
+/**
+ * Activates the action button with the enter or space key.
+ *
+ * @param {KeyboardEvent} event
+ */
+function actionButtonKeyupHandler (event) {
+  // If either enter or space is pressed, activate the button
+  if (event.keyCode === 13 || event.keyCode === 32) {
```

Comment:
The <kbd>Enter</kbd> check should be removed here. It's causing `activateActionButton` to trigger twice on <kbd>Enter</kbd> keypresses, effectively making the toggle button pointless.

Reply:
Oh, what a blunder. Of course.